substra/sdk/client.py

```diff
@@ -52,40 +52,77 @@ def wrapper(*args, **kwargs):
 
 
 class Client(object):
-    """Create the client.
+    """Create a client
 
-    Set the `backend` to 'local' to use the local debugging.
-    One client corresponds to one profile.
+    Args:
+
+        url (str, optional): URL of the Substra platform. Mandatory
+            to connect to a Substra platform. If no URL is given and debug is True, all
+            assets must be created locally.
+            Defaults to None.
+
+        token (str, optional): Token to authenticate to the Substra platform.
+            If no token is given, use the 'login' function to authenticate.
+            Defaults to None.
+
+        retry_timeout (int, optional): Number of seconds before attempting a retry call in case
+            of timeout.
+            Defaults to 5 minutes.
+
+        version (str, optional): API version, do not change it. Defaults to '0.0'.
+
+        insecure (bool, optional): If True, the client can call a not-certifed backend. This is
+            for debugging purposes.
+            Defaults to False.
+
+        debug (bool, optional): Whether to use the default or debug mode.
+            In debug mode, new assets are created locally but can access assets from
+            the deployed Substra platform. The platform is in read-only mode.
+            Defaults to False.
     """
 
     def __init__(
         self,
         url: typing.Optional[str] = None,
         token: typing.Optional[str] = None,
         retry_timeout: int = DEFAULT_RETRY_TIMEOUT,
-        backend: str = 'remote',
         version: str = '0.0',
         insecure: bool = False,
+        debug: bool = False,
     ):
-        self._backend_name = backend
         self._retry_timeout = retry_timeout
         self._token = token
 
         self._insecure = insecure
         self._url = url
         self._version = version
 
-        self._backend = self._get_backend()
-
-    def _get_backend(self):
-        return backends.get(
-            self._backend_name,
-            url=self._url,
-            version=self._version,
-            insecure=self._insecure,
-            token=self._token,
-            retry_timeout=self._retry_timeout,
-        )
+        self._backend = self._get_backend(debug)
+
+    def _get_backend(self, debug: bool):
+        # Three possibilites:
+        # - debug is False: get a remote backend
+        # - debug is True and no url is defined: fully local backend
+        # - debug is True and url is defined: local backend that connects to
+        #                           a remote backend (read-only)
+        backend = None
+        if (debug and self._url) or not debug:
```

Comment:
Could it be simplified to `if self._url` ?

Reply:
It would work, I made this condition to keep the same behaviour as before in the 'normal' (ie not debug) case.  

If the condition is `if self._url:`, then the client is created without error but any subsequent call fails with a NoneType error (since client._backend is None)

With the condition `if (debug and self._url) or not debug:`, in normal mode if there is no URL the client creation fails with an explicit error `substra.sdk.exceptions.SDKException: url required to connect to the Substra server`

Reply:
Thanks for the explanation. Looks good :)